### ISSUE TITLE
data-grammar too restrictive

### ISSUE THREAD

**Issue by KennethPierce** [atom-haskell/ide-haskell#98]
The auto grammar selector for my files is selecting "source hs" instead of source haskell. This causes no markup on the file as well as the haskell ide menu's check and lint commands to be ignored. I'd propose the following 
1) the haskell-ide menu isn't shown if the wrong data-grammar is there
2) the check and lint buttons should give an error if nothing is bound to them.
3) file ide-haskell.coffee: `atom.commands.add 'atom-text-editor[data-grammar~="haskell"]',` this line should be changed to allow for hs.

example of my data-grammar-
`<atom-text-editor class="editor" tabindex="-1" data-grammar="source hs" data-encoding="utf8" style="display: none;"></atom-text-editor>`

to repro: you can manually change your grammar using the ctrl-shift-L with the grammar-selector package.  My selector has 4 haskell choices (Haskell,Haskell,Haskell (happy),Haskell literate)

**Comment by lierdakil** [comment]
We only support language-haskell as Haskell grammar package. Autocompletion relies heavily on environments defined by specifically language-haskell. Supporting other grammar packages would be nigh impossible.

TL;DR Please consider disabling haskell-grammar package, as it's not supported.

**Comment by KennethPierce** [comment]
Thanks,
Disabling the haskell-grammar package fixed it!

That incompatibility wasn't obvious - maybe the README.md should recommend
disabling it in the requirements section.

**Comment by lierdakil** [comment]
Well, installing several language packages for the same language is not a very good idea in the first place, unless you can control which is used when (i.e. you know what you're doing), and readme specifically points out that ide-haskell uses language-haskell... I might add a note that installing several language packages for Haskell can lead to unexpected behavior. Would that be agreeable?

**Comment by KennethPierce** [comment]
That would probably be sufficient.

Atom doesn't seem to have a better solution for it.
<https://discuss.atom.io/t/what-is-good-etiquette-for-packages-whose-functionality-collides-with-another-package/7571>
